### PR TITLE
Fix bug in SQSTransformation

### PR DESCRIPTION
**Pull request by esoteric-ephemera**
The `SQSTransformation` class in `pymatgen.transformations.advanced_transformations` has a `return_ranked_list` kwarg in `apply_transformation`. 

Because this can be either boolean (return all best SQS structures from parallel runs) or an int (return only the `return_ranked_list` best SQS structures) and `bool` subclasses `int`in python, a previous check using `isinstance(return_ranked_list,int)` lead to returning only the first element in the ranked structure list (`int(True)=1`) rather than all structures in the list.

Fixed this bug and added to the existing TestSQSTransformation class to ensure this is checked.

### CODE REVIEW

**Comment by mkhorton** [comment]
Thanks Aaron!

**Comment by mkhorton** [comment]
Also noting another (Pythonic) SQS generator in `icet` although I have not tried it myself.

**Review by janosh — APPROVED** [review]
Thanks a lot @esoteric-ephemera! 👍

File: pymatgen/transformations/advanced_transformations.py [hunk 1]
```diff
@@ -1874,31 +1874,33 @@ def _proj(b, a):
...
-                       scaling=4 would lead to a 32 atom supercell
-                    b. A sequence of three scaling factors, e.g., [2, 1, 1], which
-                       specifies that the supercell should have dimensions 2a x b x c
+                a. (preferred) Scales number of atoms, e.g., for a structure with 8 atoms,
+                    scaling=4 would lead to a 32 atom supercell
+                b. A sequence of three scaling factors, e.g., [2, 1, 1], which
+                    specifies that the supercell should have dimensions 2a x b x c
             cluster_size_and_shell (Optional[Dict[int, int]]): Dictionary of cluster interactions with entries in
                 the form number of atoms: nearest neighbor shell
         Keyword Args:
```
Comment (janosh) [inline pymatgen/transformations/advanced_transformations.py:1906]: @esoteric-ephemera Could you check if this doc string is incorrect?

> Currently supported options are "niggli" or "LLL". "False" does not reduce structure.

Does `"False"` actually changes the behavior? Don't have ATAT installed myself.

Comment (esoteric-ephemera) [reply pymatgen/transformations/advanced_transformations.py:1906]: The structure reduction is actually done through PMG's `IStructure.get_reduced_structure` function, which only supports "niggli" or "LLL". Passing "False" would return a ValueError, but passing `False` (bool) would skip structure reduction. That docstr should probably be amended

Comment (janosh) [reply pymatgen/transformations/advanced_transformations.py:1906]: will do

**Pull request merged**